apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java

```diff
@@ -38,22 +69,805 @@ public HHAClaim generateClaim() throws FHIRException {
+            // valueQuantity
+            .setValue(new Quantity().setValue(3));
+
+    Assert.assertTrue(compare.equalsDeep(sic));
+  }
+
+  @Test
+  public void shouldHaveAdmissionPeriodSupInfo() throws Exception {
+    SupportingInformationComponent sic =
+        TransformerTestUtilsV2.findSupportingInfoByCode("admissionperiod", eob.getSupportingInfo());
+
+    SupportingInformationComponent compare =
+        TransformerTestUtilsV2.createSupportingInfo(
+            // We don't care what the sequence number is here
+            sic.getSequence(),
+            // Category
+            Arrays.asList(
+                new Coding(
+                    "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBSupportingInfoType",
+                    "admissionperiod",
+                    "Admission Period")));
+    // timingPeriod
+    Period period = new Period();
+    period.setStart(
+        new SimpleDateFormat("yyy-MM-dd").parse("2015-06-23"), TemporalPrecisionEnum.DAY);
+
+    compare.setTiming(period);
+
+    Assert.assertTrue(compare.equalsDeep(sic));
+  }
+
+  /** Diagnosis elements */
+  @Test
+  public void shouldHaveDiagnosesList() {
+    Assert.assertEquals(4, eob.getDiagnosis().size());
+  }
+
+  @Test
+  public void shouldHaveDiagnosesMembers() {
+    DiagnosisComponent diag1 =
+        TransformerTestUtilsV2.findDiagnosisByCode("H5555", eob.getDiagnosis());
+
+    DiagnosisComponent cmp1 =
+        TransformerTestUtilsV2.createDiagnosis(
+            // Order doesn't matter
+            diag1.getSequence(),
+            new Coding("http://hl7.org/fhir/sid/icd-9-cm", "H5555", null),
+            new Coding(
+                "http://terminology.hl7.org/CodeSystem/ex-diagnosistype", "principal", "principal"),
+            null,
+            null);
+
+    Assert.assertTrue(cmp1.equalsDeep(diag1));
+
+    DiagnosisComponent diag2 =
+        TransformerTestUtilsV2.findDiagnosisByCode("H8888", eob.getDiagnosis());
+
+    DiagnosisComponent cmp2 =
+        TransformerTestUtilsV2.createDiagnosis(
+            // Order doesn't matter
+            diag2.getSequence(),
+            new Coding("http://hl7.org/fhir/sid/icd-10", "H8888", null),
+            new Coding(
+                "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBClaimDiagnosisType",
+                "secondary",
+                "secondary"),
+            null,
+            null);
+
+    Assert.assertTrue(cmp2.equalsDeep(diag2));
+
+    DiagnosisComponent diag3 =
+        TransformerTestUtilsV2.findDiagnosisByCode("R2222", eob.getDiagnosis());
+
+    DiagnosisComponent cmp3 =
+        TransformerTestUtilsV2.createDiagnosis(
+            // Order doesn't matter
+            diag3.getSequence(),
+            new Coding("http://hl7.org/fhir/sid/icd-10", "R2222", null),
+            new Coding(
+                "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBClaimDiagnosisType",
+                "secondary",
+                "secondary"),
```

Comment:
Same issue where Display needs to be Capitalized.

Reply:
I was following what I saw in the JSON. When I capitalize the display name, the test fails

Reply:
Then it may well be that we have a minor bug in the Transformer code; to wit:

http://hl7.org/fhir/us/carin-bb/STU1/CodeSystem-C4BBClaimDiagnosisType.html

So that is essentially a 'base' code system; as noted in the above reference (URL), we see the following:

This Code system is referenced in the content logical definition of the following value sets:

C4BBClaimInpatientInstitutionalDiagnosisType
C4BBClaimOutpatientInstitutionalDiagnosisType
C4BBClaimProfessionalAndNonClinicianDiagnosisType

Now while the Display attribute is not that bif a deal, the fact that we reference
http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBClaimDiagnosisType
but then don't follow the spec, is effectively an error.

Reply:
I see. Yes. Seeing how the maven build isnt passing, Ill push this suggested change and we can address the local error.